app/src/main/java/com/example/lhi06/residue/FindAlbumActivity.java

```diff
@@ -1,36 +1,66 @@
 package com.example.lhi06.residue;
 
+import android.os.Bundle;
 import android.support.annotation.NonNull;
 import android.support.v7.app.AppCompatActivity;
-import android.os.Bundle;
 import android.support.v7.widget.LinearLayoutManager;
 import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.widget.EditText;
 
 import com.squareup.picasso.Picasso;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.io.IOException;
 
 import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class FindAlbumActivity extends AppCompatActivity {
 
     AlbumAdapter adapter;
+    ItunesService service;
 
-    private List<Album> listAlbum = new ArrayList<>();
 
     @BindView(R.id.recycleView) RecyclerView recyclerView;
+    @BindView(R.id.request) EditText editText;
 
     @Override
     protected void onCreate(@NonNull final Bundle savedInstanceState) {
```

Comment:
If you annotate a parameter as @NonNull then it means that parameter should never be null - if you try to call the function and pass in a value that can be null it should generate a compile time warning. @Nullable means the parameter can be null and Android Studio should warn you if you try to access the parameter object without doing a null check first.